web/src/js/__tests__/ducks/utils/storeSpec.js

```diff
@@ -0,0 +1,77 @@
+jest.unmock('../../../ducks/utils/store')
+
+import reduceStore, * as storeActions from '../../../ducks/utils/store'
+
+describe('store reducer', () => {
+    it('should return initial state', () => {
+        expect(reduceStore(undefined, {})).toEqual({
+            byId: {},
+            list: [],
+            listIndex: {},
+            view: [],
+            viewIndex: {},
+        })
+    })
+
+    it('should handle add action', () => {
+        let state = reduceStore(undefined, {})
+        expect(state = reduceStore(state, storeActions.add({id: 1}))).toEqual({
+                byId: { [1]: {id: 1} },
+                listIndex: { [1]: 0 },
+                list: [ {id: 1} ],
+                view: [ {id: 1} ],
+                viewIndex: { 1: 0 },
+        })
+
+        expect(reduceStore(state, storeActions.add({id: 9}))).toEqual({
+            byId: { [1]: {id:1}, [9]: {id:9} },
+            listIndex: { [1]: 0, [9]: 1 },
+            list: [ {id: 1}, {id: 9} ],
+            view: [ {id: 1}, {id: 9} ],
+            viewIndex: { 1: 0, 9: 1 },
+        })
+    })
+
+    it('should not add the item with duplicated id', () => {
+        let state = reduceStore(undefined, storeActions.add({id: 1}))
+        expect(reduceStore(state, storeActions.add({id: 1}))).toEqual(state)
+    })
+
+    it('should handle update action', () => {
+        let state = reduceStore(undefined, storeActions.add({id: 1, foo: "foo"}))
+        expect(reduceStore(state, storeActions.update({id:1, foo:"foo1"}))).toEqual({
```

Comment:
Appreciate for the advices, these will definitely make our code more readable. Now I get a better concept about how a good test case should be.